src/Http/Controllers/ImportController.php

```diff
@@ -37,46 +36,67 @@ public function preview(NovaRequest $request, $file)
 
         $sample = $import->take(10)->all();
 
-        $resources = collect(Nova::$resources);
-
-        $resources = $resources->filter(function ($resource) {
-            if ($resource === ActionResource::class) {
-                return false;
-            }
+        $resources = $this->getAvailableResourcesForImport($request); 
 
-            if (!isset($resource::$model)) {
-                return false;
-            }
+        $fields = $resources->mapWithKeys(function ($resource) use ($request) {
+            return $this->getAvailableFieldsForImport($resource, $request);
+        });
 
-            $static_vars = (new \ReflectionClass((string) $resource))->getStaticProperties();
+        $resources = $resources->mapWithKeys(function ($resource) {
+            return [$resource::uriKey() => $resource::label()];
+        });
 
-            if(!isset($static_vars['canImportResource'])) {
-                return true;
-            }
+        return response()->json(compact('sample', 'resources', 'fields', 'total_rows', 'headings'));
+    }
 
-            return isset($static_vars['canImportResource']) && $static_vars['canImportResource'];
-        });
+    public function getAvailableFieldsForImport(String $resource, $request)
+    {
+        $novaResource = new $resource(new $resource::$model);
+        $fieldsCollection = collect($novaResource->creationFields($request));
 
-        $fields = $resources->map(function ($resource) {
-            $model = $resource::$model;
+        if (method_exists($novaResource, 'exceptAttributesImportResource')) {
+            $fieldsCollection = $fieldsCollection->filter(function(Field $field) use ($novaResource, $request){
+                return !in_array($field->attribute, $novaResource->exceptAttributesImportResource($request));
+            });
+        }
```

Comment:
Consider renaming this method

```suggestion
        if (method_exists($novaResource, 'excludeAttributesFromImport')) {$fieldsCollection = $fieldsCollection->filter(function(Field $field) use ($novaResource, $request) {
                return !in_array($field->attribute, $novaResource->exludeAttributesFromImport($request));
            });
        }
```